src/mdio/converters/mdio.py

```diff
@@ -138,8 +148,12 @@ def mdio_to_segy(  # noqa: PLR0912, PLR0913
         dim_slices += (slice(start, stop),)
 
     # Lazily pull the data with limits now, and limit mask so its the same shape.
-    live_mask, headers, samples = mdio[dim_slices]
-    live_mask = live_mask.rechunk(headers.chunks)
+    # Workaround: currently the MDIO Xarray dataset returns numpy arrays instead of Dask arrays
+    # TODO (Dmitriy Repin): Revisit after the eager memory allocation is fixed
+    # https://github.com/TGSAI/mdio-python/issues/608
+    live_mask = _get_dask_array(mdio_xr, "trace_mask", new_chunks[:-1])[dim_slices]
+    headers = _get_dask_array(mdio_xr, "headers", new_chunks[:-1])[dim_slices]
+    samples = _get_dask_array(mdio_xr, "amplitude", new_chunks)[dim_slices]
```

Comment:
We should use the key we got from the metadata.

```suggestion
    samples = _get_dask_array(mdio_xr, trace_variable_name, new_chunks)[dim_slices]
```